client/rest-high-level/src/main/java/org/elasticsearch/client/GetAliasesResponse.java

```diff
@@ -0,0 +1,176 @@
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        GetAliasesResponse that = (GetAliasesResponse) o;
+        return status == that.status &&
+                Objects.equals(errorMessage, that.errorMessage) &&
+                Objects.equals(aliases, that.aliases);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(status, errorMessage, aliases);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        {
+            if (status != RestStatus.OK) {
+                builder.field("error", errorMessage);
+                builder.field("status", status.getStatus());
+            }
+
+            for (Map.Entry<String, Set<AliasMetaData>> entry : aliases.entrySet()) {
+                builder.startObject(entry.getKey());
+                {
+                    builder.startObject("aliases");
+                    {
+                        for (final AliasMetaData alias : entry.getValue()) {
+                            AliasMetaData.Builder.toXContent(alias, builder, ToXContent.EMPTY_PARAMS);
+                        }
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    public static GetAliasesResponse fromXContent(XContentParser parser) throws IOException {
+        if (parser.currentToken() == null) {
+            parser.nextToken();
+        }
+        ensureExpectedToken(Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+        Map<String, Set<AliasMetaData>> aliases = new HashMap<>();
+
+        String currentFieldName;
+        Token token;
+        String exceptionMessage = null;
+        RestStatus status = RestStatus.OK;
+
+        while (parser.nextToken() != Token.END_OBJECT) {
+            if (parser.currentToken() == Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+
+                if ("status".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
+                        ensureExpectedToken(Token.VALUE_NUMBER, token, parser::getTokenLocation);
+                        status = RestStatus.fromCode(parser.intValue());
+                    }
+                } else if ("error".equals(currentFieldName)) {
+                    if ((token = parser.nextToken()) != Token.FIELD_NAME) {
+                        if (token == Token.VALUE_STRING) {
+                            exceptionMessage = parser.text();
+                        } else if (token == Token.START_OBJECT) {
+                            parser.nextToken();
+                            exceptionMessage = ElasticsearchException.innerFromXContent(parser, true).getMessage();
```

Comment:
it would be good to throw, but the problem is that this is in reality a 404, so for which 404s do we throw and for which we don't? We went for never throwing then, but the response doesn't have a place to hold the whole exception. Shall we add that at this point? This response is probably going to look even weirder.

Reply:
one alternative is what @olcbean tried before, that I didn't like at first: throw exception in the fromXContent when we find the complete exception, that is the only way to differentiate between the different 404 responses I think. Let me know what you prefer.

Reply:
We could save the exception as a member variable. It *is* weird but the API is weird.

Reply:
then you would either have the message or the exception?